ddtrace/tracer/option_test.go

```diff
@@ -84,20 +94,25 @@ func TestTracerOptionsDefaults(t *testing.T) {
 	})
 
 	t.Run("other", func(t *testing.T) {
+		// Set DD_ENV to ensure WithEnv overrides it.
+		os.Setenv("DD_ENV", "DD_ENV")
+		defer os.Unsetenv("DD_ENV")
```

Comment:
I keep coming back to this and thinking about a saying that "_Code is like humour: if it has to be explained, it's bad_". It feels a bit like you snuck this in here to cover for a test case that is not really what this unit test is about (thus the need for a comment).

Can we either remove this or create a separate test for it? I really like unit tests to be as isolated as possible, small and easy to understand. We can always leave complexity and breadth to integration tests.

Reply:
> It feels a bit like you snuck this in here to cover for a test case that is not really what this unit test is about 

That's accurate. I'll create a separate test case for this.